src/main/java/htsjdk/samtools/SAMRecord.java

```diff
@@ -1352,16 +1356,22 @@ private void requireUnsigned(final String tag) {
     }
 
     /**
+     * @return SAMBinaryTagAndValue may be null if
+     *          1) {@link #mAttributes} is null or
+     *          2) this record does not have the requested attribute
+     */
+    private SAMBinaryTagAndValue hasAttribute(final short tag) {
```

Comment:
This seems like it's wrong.  It should return a boolean.

Reply:
Or it could be named something else.

Reply:
renamed